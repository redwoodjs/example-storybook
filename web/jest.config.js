// More info at https://redwoodjs.com/docs/project-configuration-dev-test-build

const config = {
  rootDir: '../',
  preset: '@redwoodjs/testing/config/jest/web',
  verbose: true,
  moduleNameMapper: {
    // mock redwood magic webpack aliases
    // see: https://github.com/redwoodjs/redwood/blob/main/packages/testing/config/storybook/preview.js
    '~__REDWOOD__USER_WEB_DEFAULT_CSS': 'identity-obj-proxy',
    '~__REDWOOD__USER_STORYBOOK_PREVIEW_CONFIG':
      '<rootDir>/web/config/storybook.preview.js',
  },
  transform: {
    '^.+\\.mdx?$': '@storybook/addon-docs/jest-transform-mdx', // https://github.com/storybookjs/storybook/issues/11376#issuecomment-757417250
  },
  setupFilesAfterEnv: [`<rootDir>/web/test/mocks.js`],
}

module.exports = config
