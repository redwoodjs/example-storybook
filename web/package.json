{
  "name": "web",
  "version": "0.0.0",
  "private": true,
  "browserslist": {
    "development": [
      "last 1 version"
    ],
    "production": [
      "defaults",
      "not IE 11",
      "not IE_Mob 11"
    ]
  },
  "dependencies": {
    "@redwoodjs/forms": "0.49.0",
    "@redwoodjs/router": "0.49.0",
    "@redwoodjs/web": "0.49.0",
    "prop-types": "15.8.1",
    "react": "17.0.2",
    "react-dom": "17.0.2"
  }
}
