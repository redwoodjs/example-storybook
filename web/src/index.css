/**
 * START --- SETUP TAILWINDCSS EDIT
 *
 * `yarn rw setup ui tailwindcss` placed these imports here
 * to inject Tailwind's styles into your CSS.
 * For more information, see: https://tailwindcss.com/docs/installation#include-tailwind-in-your-css
 */
@import 'tailwindcss/base';
@import 'tailwindcss/components';
@import 'tailwindcss/utilities';
/**
 * END --- SETUP TAILWINDCSS EDIT
 */
body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
}
ul {
  list-style-type: none;
  margin: 1rem 0;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 1rem 0 0 ;
}
h1 > a {
  text-decoration: none;
  color: black;
}
button, input, label, textarea {
  display: block;
  outline: none;
}
label {
  margin-top: 1rem;
}
.error {
  color: red;
}
input.error, textarea.error {
  border: 1px solid red;
}
.form-error {
  color: red;
  background-color: lavenderblush;
  padding: 1rem;
  display: inline-block;
}
.form-error ul {
  list-style-type: disc;
  margin: 1rem;
  padding: 1rem;
}
.form-error li {
  display: list-item;
}
.flex-between {
  display: flex;
  justify-content: space-between;
}
.flex-between button {
  display: inline;
}
