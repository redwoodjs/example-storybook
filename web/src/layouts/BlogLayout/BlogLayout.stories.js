//@ts-check
import { within } from '@storybook/testing-library'
import { expect } from '@storybook/jest'
import BlogLayout from './BlogLayout'

export const generated = (args) => {
  return (
    <BlogLayout {...args}>
      <></>
    </BlogLayout>
  )
}

generated.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement)
  const heading = await canvas.findByRole('heading')
  expect(heading).toHaveTextContent('Redwood Blog')
  const aboutLink = await canvas.findByRole('link', { name: 'About' })
  expect(aboutLink).toHaveTextContent('About')
}

export default {
  title: 'Layouts/BlogLayout',
  parameters: {
    layout: 'fullscreen',
  },
}
