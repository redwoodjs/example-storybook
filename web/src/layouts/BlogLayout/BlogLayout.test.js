import { render } from '@redwoodjs/testing/web'

import BlogLayout from './BlogLayout'

//   Improve this test with help from the Redwood Testing Doc:
//   https://redwoodjs.com/docs/testing#testing-pages-layouts

describe('BlogLayout', () => {
  it('renders successfully', () => {
    expect(() => {
      render(<BlogLayout />)
    }).not.toThrow()
  })

  it('fragment matches snapshot', () => {
    const { asFragment } = render(<BlogLayout />)
    expect(asFragment()).toMatchSnapshot()
  })
})
