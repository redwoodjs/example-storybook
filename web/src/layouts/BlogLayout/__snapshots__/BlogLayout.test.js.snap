// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogLayout fragment matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="relative flex justify-between items-center py-4 px-8 bg-blue-700 text-white"
  >
    <h1
      class="text-5xl font-semibold tracking-tight"
    >
      <a
        class="text-blue-400 hover:text-blue-100 transition duration-100"
        href="/"
      >
        Redwood Blog
      </a>
    </h1>
    <nav>
      <ul
        class="relative flex items-center font-light"
      >
        <li>
          <a
            class="py-2 px-4 hover:bg-blue-600 transition duration-100 rounded"
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            class="py-2 px-4 hover:bg-blue-600 transition duration-100 rounded"
            href="/contact"
          >
            Contact
          </a>
        </li>
        <li>
          <a
            class="py-2 px-4"
            href="/login"
          >
            Login
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="max-w-4xl mx-auto p-12 bg-white shadow rounded-b"
  />
</DocumentFragment>
`;
